### PR TITLE
feat(HTTP Request Node): Move from Binary Buffer to Binary streaming

### PR DESCRIPTION
**Pull request by agobrech**

### CODE REVIEW

File: packages/nodes-base/nodes/HttpRequest/V3/HttpRequestV3.node.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
+import type { Readable } from 'form-data';
```
Comment (netroy) [inline packages/nodes-base/nodes/HttpRequest/V3/HttpRequestV3.node.ts:1]: this import should be from `stream` and not `form-data`

File: packages/core/src/NodeExecuteFunctions.ts [hunk 2]
```diff
@@ -689,7 +690,9 @@ async function proxyRequestToAxios(
 			if (Buffer.isBuffer(responseData)) {
 				responseData = responseData.toString('utf-8');
 			}
-			error.message = `${response.status as number} - ${JSON.stringify(responseData)}`;
+			error.message = `${response.status as number} - ${JSON.stringify(
+				binaryToBuffer(responseData),
```
Comment (netroy) [inline packages/core/src/NodeExecuteFunctions.ts:694]:
```suggestion
				await binaryToBuffer(responseData),
```

File: packages/nodes-base/nodes/HttpRequest/V3/HttpRequestV3.node.ts [hunk 3]
```diff
@@ -1129,10 +1135,25 @@ export class HttpRequestV3 implements INodeType {
...
 					this.helpers.assertBinaryData(itemIndex, inputDataFieldName);
-					requestOptions.body = await this.helpers.getBinaryDataBuffer(
-						itemIndex,
-						inputDataFieldName,
-					);
+
+					let uploadData: Buffer | Readable;
+					let contentLength: number;
+
+					const itemBinaryData = items[itemIndex].binary![inputDataFieldName];
```
Comment (netroy) [inline packages/nodes-base/nodes/HttpRequest/V3/HttpRequestV3.node.ts:1142]: Delete line 1137, and change this to 

```suggestion
					const itemBinaryData = this.helpers.assertBinaryData(itemIndex, inputDataFieldName);
```

**Review by netroy — DISMISSED** [review]
besides the 2 comments, the code LGTM

File: pnpm-lock.yaml [hunk 4]
```diff
@@ -6545,7 +6545,7 @@ packages:
       vite: ^3.0.0 || ^4.0.0
       vue: ^2.7.0-0
     dependencies:
-      vite: 4.0.4_sass@1.58.0
+      vite: 4.0.4_sass@1.55.0+terser@5.16.1
```
Comment (netroy) [inline pnpm-lock.yaml:6548]: not sure where this change is coming from.

Comment (agobrech) [reply pnpm-lock.yaml:6548]: Ran pnpm install after removing the concast imports from nodes-base

File: packages/core/src/NodeExecuteFunctions.ts [hunk 5]
```diff
@@ -689,7 +690,9 @@ async function proxyRequestToAxios(
 			if (Buffer.isBuffer(responseData)) {
 				responseData = responseData.toString('utf-8');
 			}
-			error.message = `${response.status as number} - ${JSON.stringify(responseData)}`;
+			error.message = `${response.status as number} - ${JSON.stringify(
+				await binaryToBuffer(responseData),
+			)}`;
 		}
```
Comment (maspio) [inline packages/core/src/NodeExecuteFunctions.ts:696]: Are we sure we want to do `binaryToBuffer(responseData)` here? We already stringify Buffer above and we should have a string ultimately. Am I missing something?

Comment (netroy) [reply packages/core/src/NodeExecuteFunctions.ts:696]: we should either remove the lines 690-692, or replace it with `responseData = await binaryToBuffer(responseData)`.

**Review by maspio — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@0.219.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.219.1)

**Pull request merged**